colcon_core/task/__init__.py

```diff
@@ -299,3 +300,24 @@ def install(args, rel_src, rel_dst):
             shutil.rmtree(dst)
         if not os.path.exists(dst):
             os.symlink(src, dst)
+
+
+def collect_existing_environment_hooks(path):
+    """
+    Get a list of all existing environment hooks in the given directory.
+
+    :param Path path: The path to check for existing environment hooks
+    :rtype: list
+    """
+    additional_hooks = []
+    shell_extensions = get_shell_extensions()
+    file_extensions = []
+    for shell_extensions_same_prio in shell_extensions.values():
+        for shell_extension in shell_extensions_same_prio.values():
+            file_extensions += shell_extension.get_file_extensions()
+    for file_extension in file_extensions:
```

Comment:
Is it possible to have duplicate file extensions (ie. should we guard against this)?

Reply:
No.